src/codegen/x86_64/arch_x86_64.c

```diff
@@ -1509,6 +1486,8 @@ static size_t interfering_regs(IRInstruction *instruction) {
   return mask >> 1;
 }
 
+// TODO: This should probably be used by every backend, so it should
+// move "up" somewhere.
```

Comment:
I suppose this could either go in `intermediate_representation.h` or `codegen.h`. That or some new backend header, but I feel like name mangling is more intrinsic to the language/IR than to the backends...

Reply:
True; I feel like extracting the mangling bits into their own files is very good, that way we can compile it separately into a mangler/demangler program, if we want one of those. All it'd take is some simple command line parsing and then calls to this API